src/gluonts/sagemaker/model.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+
+# Third-party imports
+from sagemaker.model import FrameworkModel
+from sagemaker.predictor import (
+    RealTimePredictor,
+    json_serializer,
+    json_deserializer,
+)
+from sagemaker import session
+from typing import List, Optional, Tuple, Dict
+
+# First-party imports
+from gluonts.sagemaker.defaults import GLUONTS_VERSION
+
+
+class GluonTSPredictor(RealTimePredictor):
+    """
+    A RealTimePredictor for inference against GluonTS Endpoints.
+    This is able to serialize and deserialize datasets in the gluonts data format.
+    """
+
+    def __init__(
+        self, endpoint_name: str, sagemaker_session: session.Session = None
+    ):
+        """Initialize an ``GluonTSPredictor``.
+        Parameters
+        ----------
+            endpoint_name:
+                The name of the endpoint to perform inference on.
+            sagemaker_session :
+                Session object which manages interactions with Amazon SageMaker APIs and any other
+                AWS services needed. If not specified, the estimator creates one
+                using the default AWS configuration chain.
+        """
+
+        # Use the default functions from MXNet (they handle more than we need
+        # (e.g: np.ndarrays), but that should be fine)
+        super(GluonTSPredictor, self).__init__(
+            endpoint_name,
+            sagemaker_session,
+            json_serializer,
+            json_deserializer,
+        )
+
+
+class GluonTSModel(FrameworkModel):
+    """An GluonTS SageMaker ``Model`` that can be deployed to a SageMaker ``Endpoint``."""
+
+    __framework_name__ = "gluonts"
+    _LOWEST_MMS_VERSION = "0.3.3"  # TODO: figure out the meaning of this...
```

Comment:
MMS is a Model Server for Apache MXNet: https://github.com/awslabs/mxnet-model-server

Reply:
Thx! I guess I will put it back to its original value of 1.4 then, we will have to see how to make proper use of it though.  Originally I thought it has to do with the gluonts version directly because of:

```
is_mms_version` = parse_version(self.framework_version) >= parse_version(
            self._LOWEST_MMS_VERSION
        )
```

But originally __framework_name__ was also "mxnet"...